packages/react-google-maps-api/src/GoogleMap.tsx

```diff
@@ -104,6 +110,7 @@ export interface GoogleMapProps {
   onZoomChanged?: () => void;
   onLoad?: (map: google.maps.Map) => void | Promise<void>;
   onUnmount?: (map: google.maps.Map) => void | Promise<void>;
+  panTo?: google.maps.LatLng | google.maps.LatLngLiteral;
```

Comment:
Shouldn't `panTo` be a function type, accepting an argument and returning `void`?

Reply:
Maybe the name is confusing. The idea is to pass a latlang, that whenever it changes that map will panTo it. I thought using the panTo name would make it clear that this uses the panTo functionality.

Reply:
the name is confusing.

Reply:
Maybe `panToCenter` will be less confusing and sounds more meaningful in a list with `center` prop?

Reply:
maybe rename panTo to autopanCoordinates ?

Reply:
I assume that user will prefer to use only one type of center change - instant (`center`) or panned (`???newPropName???`) but using both in same time could potentially cause to undesired effects - because methods `setCenter` and `panTo` called if only prop changed. I must insist to use word `center` as suffix or prefix of propname to highlight its potential problems with `center` prop used simultaneously. Or we could throw exception if user uses both, but I could imagine the case when both could be useful.

example: https://codesandbox.io/s/markers-playbox-rodck - top and bottom markers using `panTo` methods, middle uses `setCenter`. try click top, middle, bottom, middle, top - second middle click won't move the camera because its not changed `center` prop passed to component. it's could be confusing when use both props.

What are your thoughts, guys?

Reply:
@uriklar mention

Reply:
I'm pretty much settled to do not expand an API, and to do not overcomplicate things. Can't say yes for every feature, sorry. I feel that it could bite performance in general, especially for people who do not use eslint-plugin-react-perf. Providing objects as props to components is not good idea in general, and we are providing objects cos `react-google-maps` requires to. I feel that it is better to leave it as it is, and use ```const map=useGoogleMap(); map.setCenter({.....});``` as @xalisys  proposed, without drawbacks to performance for everybody. Everybody who has same requirements will google up this issue and use it.

Reply:
@JustFly1984 
couple of questions:
1.  So you are planning to remove\refactor `center` prop in future because it could affect performance too? It have analogous problem. With all the respect to you "performance problem" sounds for me like the non-logic excuse.
2. Where `useGoogleMap`  should be used? Docs have not clear example. It exposes map instance to map children throught context? How to use it in parent component?

Reply:
@JustFly1984 Yeah it is a good question, how can i get center in map events?

```js
<GoogleMap
    onBoundsChanged={() => {
        // no reference to `map` or` center` is available
    }}
  >
    {...}
</GoogleMap>
```